### PR TITLE
Implement plotting of non-time based data

### PR DESCRIPTION
**Pull request by jini-zh**

### CODE REVIEW

**Comment by jini-zh** [comment]
This pull request should be merged together with ToolDAQ/ToolDAQFramework#23, ToolDAQ/middleman#6, ToolDAQ/libDAQInterface#9

File: cgi-bin/Makefile [hunk 1]
```diff
@@ -1,15 +1,27 @@
...
-	g++ -O3  sendcommand2.cpp -o sendcommand2.cgi -L /opt/ToolDAQFramework/lib -L /opt/ToolFrameworkCore/lib -lStore -lServiceDiscovery -lDataModelBase -lDAQDataModelBase -lDAQStore -lcgicc  -I /usr/local/include -I /opt/ToolDAQFramework/include -I /opt/ToolFrameworkCore/include -I /opt/boost_1_66_0/install/include/  -I /opt/zeromq-4.0.7/include -L /opt/boost_1_66_0/install/lib  -lboost_date_time -lboost_serialization  -lboost_iostreams  -L /opt/zeromq-4.0.7/lib -lzmq 
-	chmod a+x sendcommand2.cgi
+	$(CXX) -o $@ $< $(CXXFLAGS) -lzmq -lcgicc -lStore
 
 sendcommand2nopadding.cgi: sendcommand2nopadding.cpp
-	g++ -O3  sendcommand2nopadding.cpp -o sendcommand2nopadding.cgi -L /opt/ToolDAQFramework/lib -L /opt/ToolFrameworkCore/lib -lStore -lServiceDiscovery  -lDataModelBase -lDAQDataModelBase -lDAQStore -lcgicc  -I /usr/local/include -I /opt/ToolDAQFramework/include -I /opt/ToolFrameworkCore/include -I /opt/boost_1_66_0/install/include/  -I /opt/zeromq-4.0.7/include -L /opt/boost_1_66_0/install/lib  -lboost_date_time -lboost_serialization  -lboost_iostreams  -L /opt/zeromq-4.0.7/lib -lzmq 
-	chmod a+x sendcommand2nopadding.cgi
+	$(CXX) -o $@ $< $(CXXFLAGS) -lzmq -lcgicc -lStore
```
Comment (brichards64) [inline cgi-bin/Makefile:21]: need chmod

File: cgi-bin/Makefile [hunk 2]
```diff
@@ -1,15 +1,27 @@
...
+programs = getplot.cgi sendcommand2.cgi sendcommand2nopadding.cgi
+
+.PHONY: all clean
+
+all: $(programs)
 
 sendcommand2.cgi: sendcommand2.cpp
-	g++ -O3  sendcommand2.cpp -o sendcommand2.cgi -L /opt/ToolDAQFramework/lib -L /opt/ToolFrameworkCore/lib -lStore -lServiceDiscovery -lDataModelBase -lDAQDataModelBase -lDAQStore -lcgicc  -I /usr/local/include -I /opt/ToolDAQFramework/include -I /opt/ToolFrameworkCore/include -I /opt/boost_1_66_0/install/include/  -I /opt/zeromq-4.0.7/include -L /opt/boost_1_66_0/install/lib  -lboost_date_time -lboost_serialization  -lboost_iostreams  -L /opt/zeromq-4.0.7/lib -lzmq 
-	chmod a+x sendcommand2.cgi
+	$(CXX) -o $@ $< $(CXXFLAGS) -lzmq -lcgicc -lStore
```
Comment (brichards64) [inline cgi-bin/Makefile:18]: need chmod

File: Dockerfile.server [hunk 3]
```diff
@@ -54,7 +54,7 @@ RUN echo "unalias cp" >>  /etc/rc.local ;\
     echo "chmod a+x /web/SetupDatabase.sh" >>  /etc/rc.local ;\
     echo "chmod a+x /opt/middleman/run_middleman.sh" >>  /etc/rc.local ;\
     echo "chmod a+x /opt/middleman/Setup.sh" >>  /etc/rc.local ;\
-    echo "export LD_LIBRARY_PATH=/lib/:/opt/ToolFrameworkCore/lib:/opt/ToolDAQFramework/lib:/opt/boost_1_66_0/install/lib:/opt/zeromq-4.0.7/lib:$LD_LIBRARY_PATH" >> /etc/rc.local ;\
+    echo "export LD_LIBRARY_PATH=/lib/:/opt/ToolFrameworkCore/lib:/opt/ToolDAQFramework/lib:/opt/boost_1_66_0/install/lib:/opt/zeromq-4.0.7/lib:/opt/libpqxx-6.4.5/install/lib:$LD_LIBRARY_PATH" >> /etc/rc.local ;\
```
Comment (brichards64) [inline Dockerfile.server:57]: why is this needed

File: cgi-bin/getplot.cpp [hunk 4]
```diff
@@ -0,0 +1,118 @@
+#include <iostream>
+#include <map>
+#include <string>
+#include <exception>
+
+#include <pqxx/connection>
```
Comment (brichards64) [inline cgi-bin/getplot.cpp:6]: we need to discuss this and see if it should be going direct.

**Review by brichards64 — CHANGES_REQUESTED** [review]
some changes, need to discuss the direct thingwith marcus

**Comment by jini-zh** [comment]
Should I add `GetPlot` to `html-Detector/include/functions.js`? If yes, we need a way to avoid duplicating JS code otherwise it will inevitably diverge.

**Pull request merged**